src/cljs/rems/administration/create_form.cljs

```diff
@@ -97,66 +68,62 @@
                                (focus/scroll-to-top element)
                                (.focus (.querySelector element selector))))))
 
-(rf/reg-sub ::form (fn [db _]
-                     (-> (::form db)
-                         (update :form/fields #(vec (map-indexed (fn [i field] (assoc field :field/index i)) %))))))
+(defn- assign-field-index [form]
+  (update form :form/fields #(vec (map-indexed (fn [i field] (assoc field :field/index i)) %))))
+
+(rf/reg-sub ::form-data (fn [db] (assign-field-index (get-in db [::form :data]))))
 (rf/reg-sub ::form-errors (fn [db _] (::form-errors db)))
-(rf/reg-sub ::loading-form? (fn [db _] (::loading-form? db)))
 (rf/reg-sub ::edit-form? (fn [db _] (::edit-form? db)))
-(rf/reg-sub ::organization-read-only? (fn [db _] (::organization-read-only? db)))
-(rf/reg-event-db ::set-form-field (fn [db [_ keys value]] (assoc-in db (concat [::form] keys) value)))
-
-(rf/reg-sub ::selected-organization (fn [db _] (get-in db [::form :form/organization])))
-(rf/reg-event-db ::set-selected-organization (fn [db [_ organization]] (assoc-in db [::form :form/organization] organization)))
+(rf/reg-event-db ::set-form-field (fn [db [_ keys value]] (assoc-in db (concat [::form :data] keys) value)))
 
 (rf/reg-event-db
  ::add-form-field
  (fn [db [_ & [index]]]
-   (let [new-item (merge (generate-field-id (get-in db [::form :form/fields]))
+   (let [new-item (merge (generate-field-id (get-in db [::form :data :form/fields]))
                          {:field/type :text})]
      (focus-field-editor! (:field/id new-item))
-     (update-in db [::form :form/fields] items/add new-item index))))
+     (update-in db [::form :data :form/fields] items/add new-item index))))
```

Comment:
hmm updating the state of the fetcher feels a bit ugly, but I admin that's how it was going even before the fetcher

Reply:
I think the correct abstraction is yet to be found. The fetcher is a concern for the cases where you edit stuff. So far that's not all the entities because we don't have editors for all of them. And some editors can have a clear command where to store the change instead of updating the loaded data.